backend/internal/system/security/jwt_authenticator.go

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package security
+
+import (
+	"net/http"
+	"strings"
+
+	sysContext "github.com/asgardeo/thunder/internal/system/context"
+	"github.com/asgardeo/thunder/internal/system/jwt"
+)
+
+// JWTAuthenticator handles authentication and authorization using JWT Bearer tokens.
+type JWTAuthenticator struct {
+	jwtService jwt.JWTServiceInterface
+}
+
+// newJWTAuthenticator creates a new JWT authenticator.
+func newJWTAuthenticator(jwtService jwt.JWTServiceInterface) *JWTAuthenticator {
+	return &JWTAuthenticator{
+		jwtService: jwtService,
+	}
+}
+
+// CanHandle checks if the request contains a Bearer token in the Authorization header.
+func (h *JWTAuthenticator) CanHandle(r *http.Request) bool {
+	authHeader := r.Header.Get("Authorization")
+	return strings.HasPrefix(authHeader, "Bearer ")
+}
+
+// Authenticate performs both authentication and authorization in a single operation.
+// It validates the JWT token and checks if the user has required scopes for the route.
+// Returns AuthenticationContext on success, or an error for auth (401) or authz (403) failures.
+func (h *JWTAuthenticator) Authenticate(r *http.Request) (*sysContext.AuthenticationContext, error) {
+	// Step 1: Extract Bearer token
+	authHeader := r.Header.Get("Authorization")
```

Comment:
We should have a constant for this header in the system package. Shall we reuse that?